projects/plugins/jetpack/extensions/blocks/blogroll/components/blogroll-appender/index.js

```diff
@@ -26,22 +29,91 @@ export default function BlogrollAppender( { subscriptions, clientId } ) {
 		setIsVisible( false );
 	};
 
-	// eslint-disable-next-line react-hooks/exhaustive-deps
-	useEffect(
-		debounce( () => {
-			const query = searchInput.toLowerCase();
+	const fetchSiteDetails = async searchQuery => {
+		if ( abortFetchSiteDetailsControllerRef.current ) {
+			abortFetchSiteDetailsControllerRef.current.abort();
+		}
+		abortFetchSiteDetailsControllerRef.current =
+			typeof AbortController === 'undefined' ? undefined : new AbortController();
+
+		const siteDetails = await fetch(
+			addQueryArgs(
+				'https://public-api.wordpress.com/rest/v1.1/sites/' + encodeURIComponent( searchQuery ),
+				{ force: 'wpcom' }
+			)
+		)
+			.then( response => {
+				if ( ! response.ok ) {
+					setResults( [] );
+					fetchSiteDetailsCache[ searchQuery ] = null;
+				} else {
+					return response.json();
+				}
+			} )
+			.then( data => {
+				if ( data ) {
+					fetchSiteDetailsCache[ searchQuery ] = data;
+					setResults( [
+						{
+							id: data?.ID,
+							description: data?.description,
+							URL: data?.URL,
+							site_icon: data?.logo?.url,
+							name: data?.name,
+						},
+					] );
+				} else {
+					setResults( [] );
+				}
+			} )
+			.catch( () => {
+				setResults( [] );
+			} );
+
+		return siteDetails;
+	};
 
-			setResults(
-				subscriptions.filter( item => {
-					const nameContainsSearch = item.name.toLowerCase().includes( query.toLowerCase() );
-					const urlContainsSearch = item.URL.toLowerCase().includes( query.toLowerCase() );
+	useEffect( () => {
+		const cancellableSearch = setTimeout( () => {
+			const searchQuery = searchInput.toLowerCase().trim();
+			if ( searchQuery.length > 0 ) {
+				const existInSubscriptions = subscriptions.filter( item => {
+					const nameContainsSearch = item.name.toLowerCase().includes( searchQuery.toLowerCase() );
+					const urlContainsSearch = item.URL.toLowerCase().includes( searchQuery.toLowerCase() );
 
 					return nameContainsSearch || urlContainsSearch;
-				} )
-			);
-		}, 250 ),
-		[ searchInput ]
-	);
+				} );
+
+				if ( checkIfValidDomain( searchQuery ) ) {
+					if ( searchQuery in fetchSiteDetailsCache ) {
+						const cachedSiteDetails = fetchSiteDetailsCache[ searchQuery ]
+							? [
+									{
+										id: fetchSiteDetailsCache[ searchQuery ]?.ID,
+										description: fetchSiteDetailsCache[ searchQuery ]?.description,
+										URL: fetchSiteDetailsCache[ searchQuery ]?.URL,
+										site_icon: fetchSiteDetailsCache[ searchQuery ]?.logo?.url,
+										name: fetchSiteDetailsCache[ searchQuery ]?.name,
+									},
+							  ]
+							: [];
+
+						setResults( cachedSiteDetails );
+						return;
+					}
+					fetchSiteDetails( searchQuery );
+				} else {
+					setResults( existInSubscriptions );
+				}
+			} else {
+				setResults( subscriptions );
+			}
+		}, 1000 );
+
+		return () => {
+			clearTimeout( cancellableSearch );
+		};
+	}, [ searchInput, subscriptions ] );
```

Comment:
These should be a hook.

Reply:
I moved the logic into the `use-get-site-details` hook